dispatcher.go

```diff
@@ -109,10 +109,13 @@ func NewDispatcher(cfg Config) Dispatcher {
 
 // convertOutbounds applys outbound middleware and creates validator outbounds
 func convertOutbounds(outbounds Outbounds, middleware OutboundMiddleware) Outbounds {
-	//TODO(apb): ensure we're not given the same underlying outbound for each RPC type
 	convertedOutbounds := make(Outbounds, len(outbounds))
 
 	for service, outs := range outbounds {
+		if outs.Unary == nil && outs.Oneway == nil {
+			panic(fmt.Sprintf("no outbound set for service %q in dispatcher", service))
```

Comment:
the first panic is [here](https://github.com/yarpc/yarpc-go/pull/536/files/b1593ee42f40ecb1ad581f3eb808be3e319f053d#diff-156ea3d2fceaaac2ae3492542a41ddb0R98) if we give the dispatcher an invalid name. Since this happens at the configuration step, I'd assume we'd want to blow up here?